ds3/ds3.py

```diff
@@ -6777,15 +6767,11 @@ class GetObjectResponse(AbstractResponse):
   
   def process_response(self, response):
     self.__check_status_codes__([200, 206])
-    localFile = None
-    if self.request.stream:
-      localFile = self.request.stream
-    else:
-      localFile = open(self.request.effective_file_name, "wb")
+    stream = self.request.stream
     if self.request.offset:
-      localFile.seek(self.request.offset, 0)
-    localFile.write(response.read())
-    localFile.close()
+      stream.seek(self.request.offset, 0)
+    stream.write(response.read())
```

Comment:
We should probably understand how this will behave.  If this does a single read and then write, we could be in trouble.  We may have to do a buffered read, and keep reading until the stream closes.  We also need to verify that we read the whole file, so we need to validate the amount we read with what the `Content-Length` header says.